packages/flutter_tools/templates/list_detail_app/lib/src/dummy_feature/dummy_item_list_view.dart.tmpl

```diff
@@ -0,0 +1,57 @@
+import 'package:flutter/material.dart';
+import 'package:{{projectName}}/src/dummy_feature/dummy_item.dart';
+import 'package:{{projectName}}/src/dummy_feature/dummy_item_details_view.dart';
+import 'package:{{projectName}}/src/settings/settings_view.dart';
+
+/// Displays a list of DummyItems.
+class DummyItemListView extends StatelessWidget {
+  const DummyItemListView({
+    Key key,
+    this.items = const [DummyItem(1), DummyItem(2), DummyItem(3)],
+  }) : super(key: key);
+
+  static const routeName = '/';
+
+  final List<DummyItem> items;
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text('Dummy Items'),
+        actions: [
+          IconButton(
+            icon: const Icon(Icons.settings),
+            onPressed: () {
+              Navigator.pushNamed(context, SettingsView.routeName);
+            },
+          ),
+        ],
+      ),
+
+      // To work with lists that may contain a large number of items, it’s best
+      // to use the ListView.builder constructor.
+      //
+      // In contrast to the default ListView constructor, which requires
+      // building all Widgets up front, the ListView.builder constructor lazily
+      // builds Widgets as they’re scrolled into view.
+      body: ListView.builder(
```

Comment:
would it make sense to use the ListView builder that gives the items a fixed height using the prototype widget?

Reply:
Sorry, I'm not quite sure I understand. Are you suggesting we add an `itemExtent` here to give items a fixed height and convert each item into `Placeholder` widget?

Reply:
I meant using SliverPrototypeExtentList. Filed https://github.com/flutter/flutter/issues/76719 because we really should make that easier.

Reply:
Ah, wow -- I'd never used that one myself! Thanks for the info :)

My question: Since this is aimed at intermediate Flutter developers, do you think it might be best to point them to the simpler `ListView.builder` for now until #76719 is addressed? 

The reason I introduced `ListView.builder` instead of a regular `ListView` was because I was actually asked about the difference during one of my user interviews and I thought it would be nice to highlight. If we use `SliverPrototypeExtentList`, we need to introduce the concepts of `CustomScrollView`, `slivers`, and `SliverChildBuilderDelegate` as well. Overall, it's doable, but might be a bit much given the target audience.

What do you think?

```dart
      // Updated version using SliverPrototypeExtentList
      CustomScrollView(
        slivers: [
          SliverPrototypeExtentList(
            delegate: SliverChildBuilderDelegate(
                  (context, index) {
                final item = items[index];

                return ListTile(
                  title: Text('DummyItem ${item.id}'),
                  leading: const CircleAvatar(
                    // Display the Flutter Logo image asset.
                    foregroundImage:
                    AssetImage('assets/images/flutter_logo.png'),
                  ),
                  onTap: () {
                    Navigator.restorablePushNamed(
                      context,
                      DummyItemDetailsView.routeName,
                    );
                  },
                );
              },
              childCount: items.length,
            ),
            prototypeItem: ListTile(),
          )
        ],
      )
```

Reply:
ListView.builder is probably fine. We can revisit when #76719 is done.